src/database/declarations.h

```diff
@@ -1,17 +1,56 @@
-#ifndef TABLECOLUMNLITERALS_H
-#define TABLECOLUMNLITERALS_H
+#ifndef DECLARATIONS_H
+#define DECLARATIONS_H
 
-#include <QString>
-#include <QStringLiteral>
+#include <QtCore>
 
-// [F]: These constants deviate slightly from const naming convention to reflect their sql column name.
+/*!
+ * \brief An alias for QString
+ *
+ * Very long description *with* **markdown?**
+ * - - -
+ * # Header
+ */
+using ColName = QString;
+using ColData = QVariant;
+using TableName = QString;
+using RowId = int;
+
+using TableNames = QStringList;
+// [G]: May lead to some confusion. TableData suggest data for the entire table.
+// but in reallity it is data per column *of single row* (unless i misunderstand)
+// [F]: That's correct. We could maybe call it EntryData or RowData?
+using RowData = QMap<ColName, ColData>;
+using ColumnData = QPair<ColName, ColData>;
+using ColumnNames = QStringList;
+using TableColumns = QMap<TableName, ColumnNames>;
+
+// [G]: Needs some work. Inheriting from QPair may be helpful but
+// may also be overkill. Lets determine the specific uses of DataPosition
+// and provide our own interface i would say.
+// [F]: Good idea! Implementing something similar to first and second methods
+// of QPair would be useful to carry over, or some other way of quickly and
+// unambiguously accessing the elements.
```

Comment:
first and second arent methods. They are members. QPair is a struct so everything is public unless specified